### PR TITLE
Extend secrets docs with key to path mapping and automatic updates

### PR DESCRIPTION
**Pull request by ingvagabund**
Secret volume plugin has been refactored. Currently, AtomicWritter is used to project all keys.
Allowing to automatically refresh projected secrets.

At the same time, operator can specify target file for each secret and thus override the default projection location.

### CODE REVIEW

**Comment by ingvagabund** [comment]
This PR depends on https://github.com/kubernetes/kubernetes/pull/25285. Once #25285 gets merged, the describe functionality will be available.

File: docs/user-guide/secrets/index.md [hunk 1]
```diff
@@ -214,6 +214,52 @@ You can package many files into one secret, or use many secrets, whichever is co
 
 See another example of creating a secret and a pod that consumes that secret in a volume [here](/docs/user-guide/secrets/).
 
+##### Projection of secret keys to specific paths
```
Comment (johndmulhausen) [inline docs/user-guide/secrets/index.md]: I think <h5>s are not really a thing on the site. :) Reduce to 4.

**Comment by johndmulhausen** [comment]
@ingvagabund #25285 looks like a 1.3-bound piece of functionality now; If you like, I can leave this open until such time as #25285 is merged or you can open this same PR against the release-1.3 branch and we can merge *that* PR right away. Upon 1.3 release time, release-1.3 will be promoted to master.

File: docs/user-guide/secrets/index.md (hunk 1 above)
Comment (ingvagabund) [inline docs/user-guide/secrets/index.md]: There are already level-5 headings on the site. At the same time, this part must go under "Using Secrets as Files from a Pod" which is level-4 heading already.

Comment (ingvagabund) [inline docs/user-guide/secrets/index.md]: What about change all level-5 headings into bold heading?

**Comment by ingvagabund** [comment]
This should be merged before before master promotion. Let's solve the level-5 heading problem first. Then, I will open the PR for release-1.3.

Updated. PTAL.

**Comment by pmorie** [comment]
@johndmulhausen the PR in kube is merged; can we merge this?  This is a significant source of questions from users.

**Comment by ingvagabund** [comment]
@johndmulhausen PR for ``release-1.3`` branch at #628

**Comment by johndmulhausen** [comment]
@pmorie @ingvagabund -- Is this documenting release-1.3 functionality? Should it currently be considered Tech Review LGTM?

**Comment by ingvagabund** [comment]
@johndmulhausen yes it is, yes it should.

**Comment by johndmulhausen** [comment]
Thanks, @ingvagabund !

**Pull request merged**